prooph cs fixer test image.

bweston92 opened the pull request:
image for prooph cs fixer #8

Commit:
prooph cs fixer test image.

prolic commented:
LGTM. @sandrokeil can you check this? You're the pro here :)

bweston92 commented:
I would drop the composer baggage as it could save a few megabytes using
the new pipeline stuff in Dockerfiles but not sure on whether the thing
that builds these images will be up to date.
On Fri, 28 Apr 2017 at 07:35, Sascha-Oliver Prolic <notifications@github.com>
wrote:

> LGTM. @sandrokeil <https://github.com/sandrokeil> can you check this?
> You're the pro here :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prooph/docker-files/pull/9#issuecomment-297919166>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABrq2Xvw5PZq2IiIB1vtSyJV25K4mztsks5r0YiUgaJpZM4NKmaL>
> .
>

codeliner commented on php-cs-test/README.md:
```diff
@@ -0,0 +1,11 @@
+
+# Usage
+
+Mount code to `/sut` on the container and we will test again prooph's rules.
```
Has `sut` a special meaning? In other containers (f.e. prooph/php-cli) we mount $(pwd) to `/app`
Would it be ok to do the same here?

sandrokeil replied:
@bweston92  Yep, please use `/app`.

bweston92 replied:
Done

bweston92 commented:
I just chose SUT for "subject under test" can do anything as long as
entrypoint is updated.
On Fri, 28 Apr 2017 at 12:28, Alexander Miertsch <notifications@github.com>
wrote:

> *@codeliner* commented on this pull request.
> ------------------------------
>
> In php-cs-test/README.md
> <https://github.com/prooph/docker-files/pull/9#discussion_r113906369>:
>
> > @@ -0,0 +1,11 @@
> +# Introduction
> +
> +The aim of this container image is to eliminate the need to `php-cs-fixer` in the composer requirements for each project we do. This image can be used on any environment that has Docker installed including a CI pipeline.
> +
> +# Usage
> +
> +Mount code to `/sut` on the container and we will test again prooph's rules.
>
> Has sut a special meaning? In other containers (f.e. prooph/php-cli) we
> mount $(pwd) to /app
> Would it be ok to do the same here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prooph/docker-files/pull/9#pullrequestreview-35334455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABrq2b_dXrhwOvk5b5Pj3XB6VOeCTEQnks5r0c1TgaJpZM4NKmaL>
> .
>

codeliner commented:
ah ok, I like it! But I think it is better to keep it consistent with the other images and mount it to `app`

bweston92 commented:
That's fine, will update.

sandrokeil commented on php-cs-test/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+FROM composer
+
```
We should use `prooph/composer:7.1` here

bweston92 replied:
I will look on the image sizes, but I'd imagine a bare composer base image would be less in size.

bweston92 replied:
I've gone ahead and used composer as the official image still as it only needs to collect the file in that step.

sandrokeil commented on php-cs-test/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+VOLUME /sut
+WORKDIR /sut
+
+ENTRYPOINT ["/composer/vendor/bin/php-cs-fixer", "fix", "--dry-run", "--diff", "--config=/rules.php"]
```
@bweston92 What do you think about these lines, so the CMD arguments can be overridden?

```
CMD ["fix", "--dry-run", "--diff"]
ENTRYPOINT ["/composer/vendor/bin/php-cs-fixer", "--config=/rules.php"]
```

The composer path is another path with `prooph/composer`.

bweston92 replied:
I can do, I was more thinking this image is the test and therefore wouldn't be run any different. Will update PR soon.

bweston92 replied:
Done

Commit:
Updated Dockerfile for php-cs-test

The dependencies are now installed using a lightweight composer image, once the dependencies has been fetched we copy them in a php image and run php-cs-fixer against /app

bweston92 commented:
@sandrokeil can you take a look at it.

Commit:
Back to using PHP's official image

Still unsure as to why I'm using the prooph/php image as it is far bigger then php's official image. It's not like the php-cs-fixer needs anything more then what is already available with php's official image.

bweston92 commented:
/ping @sandrokeil

Commit:
Moved rules to test `/app`

Commit:
Update volume path on README
